utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -14,46 +14,106 @@
-    this->windowEnd_ = util::DateTime(winend);
-
-    // get input netcdf files
-    fullConfig.get("input files", this->inputFilenames_);
-
-    // ioda output file name
-    this->outputFilename_ = fullConfig.getString("output file");
-  }
-
-    // Method to write out a IODA file
+    // Constructor: Stores the configuration as a data members
+    explicit NetCDFToIodaConverter(const eckit::Configuration & fullConfig) {
+      // time window info
+      std::string winbegin;
+      std::string winend;
+      std::string obsvar;
+      fullConfig.get("window begin", winbegin);
+      fullConfig.get("window end", winend);
+      fullConfig.get("variable", obsvar);
+      this->windowBegin_ = util::DateTime(winbegin);
+      this->windowEnd_ = util::DateTime(winend);
+      this->variable_ = obsvar;
+      oops::Log::info() << "--- Window begin: " << winbegin << std::endl;
+      oops::Log::info() << "--- Window end: " << winend << std::endl;
+      oops::Log::info() << "--- Variable: " << obsvar << std::endl;
+
+      // get input netcdf files
+      fullConfig.get("input files", this->inputFilenames_);
+      oops::Log::info() << "--- Input files: " << this->inputFilenames_ << std::endl;
+
+      // ioda output file name
+      this->outputFilename_ = fullConfig.getString("output file");
+      oops::Log::info() << "--- Output files: " << this->outputFilename_ << std::endl;
+    }
+
+    // Method to write out a IODA file (writter called in destructor)
     void WriteToIoda() {
       // Create empty group backed by HDF file
       ioda::Group group =
         ioda::Engines::HH::createFile(
                                       this->outputFilename_,
                                       ioda::Engines::BackendCreateModes::Truncate_If_Exists);
-      this->ReadFromNetCDF(group);
+
+      // Extract ioda variables from the provider's files
+      std::string fileName = this->inputFilenames_[0];  // TODO(Guillaume): make it work for a list
+      gdasapp::IodaVars iodaVars;
+      this->ProviderToIodaVars(fileName, iodaVars);
+      oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+      oops::Log::debug() << "--- iodaVars.obsVal: " << iodaVars.obsVal << std::endl;
+
+      // Update the group with the location dimension
+      ioda::NewDimensionScales_t
+        newDims {ioda::NewDimensionScale<int>("Location", iodaVars.location)};
+      ioda::ObsGroup ogrp = ioda::ObsGroup::generate(group, newDims);
+
+      // Set up the float creation parameters
+      ioda::VariableCreationParameters float_params;
+      float_params.chunk = true;               // allow chunking
+      float_params.compressWithGZIP();         // compress using gzip
+      float missing_value = util::missingValue(missing_value);
+      float_params.setFillValue<float>(missing_value);
+
+      // Create the IODA variables
+      ioda::Variable adtIodaDatetime =
+        ogrp.vars.createWithScales<float>("Metadata/dateTime",
```

Comment:
Just wonder that is it going to be "float" instead of "int or int64"?

Reply:
nex PR @apchoiCMD